### DISCUSSION TITLE
Problem with install rosetta mpi

### DISCUSSION THREAD

**Discussion by Mikukub** [RosettaCommons/rosetta#276] [Build & Install] [answered]
I have this problem to install and get this error

mpiCC -o build/src/release/linux/4.18/64/x86/gcc/8/mpi-serialization/torsional_potential_corrections.mpiserialization.linuxgccrelease -Wl,--disable-new-dtags -Wl,-rpath=/data1/tlarnd2/software/rosetta/source/build/src/release/linux/4.18/64/x86/gcc/8/mpi-serialization -Wl,-rpath=/data1/tlarnd2/software/rosetta/source/build/external/release/linux/4.18/64/x86/gcc/8/mpi-serialization -Wl,-rpath=\$ORIGIN -Wl,-rpath=\$ORIGIN/../lib build/src/release/linux/4.18/64/x86/gcc/8/mpi-serialization/apps/public/weight_optimization/torsional_potential_corrections.o -Lexternal/lib -Lbuild/src/release/linux/4.18/64/x86/gcc/8/mpi-serialization -Lsrc -Lbuild/external/release/linux/4.18/64/x86/gcc/8/mpi-serialization -Lexternal -L/usr/local/amber22/lib -L/usr/local/cuda/lib -L/usr/local/cuda/lib64 -L/usr/local/amber22/lib -L/usr/local/cuda/lib -L/usr/local/cuda/lib64 -L/usr/local/amber22/lib -L/usr/local/cuda/lib -L/usr/local/cuda/lib64 -L/usr/local/amber22/lib -L/usr/local/cuda/lib -L/usr/local/cuda/lib64 -L/usr/lib -L/usr/local/lib -ldevel -lprotocols.8 -lprotocols.7 -lprotocols_e.6 -lprotocols_d.6 -lprotocols_c.6 -lprotocols_b.6 -lprotocols_a.6 -lprotocols_h.5 -lprotocols_g.5 -lprotocols_f.5 -lprotocols_e.5 -lprotocols_d.5 -lprotocols_c.5 -lprotocols_b.5 -lprotocols_a.5 -lprotocols.4 -lprotocols.3 -lprotocols_b.2 -lprotocols_a.2 -lprotocols.1 -lcore.6 -lcore.5 -lcore.4 -lcore.3 -lcore.2 -lcore.1 -lbasic -lnumeric -lutility -lObjexxFCL -lz -lzlib -lcppdb -lsqlite3 -lcifparse -lxml2 -lzmq -lrdkit -lbcl -lcmaes
/usr/bin/ld: warning: libmpi.so.12, needed by build/external/release/linux/4.18/64/x86/gcc/8/mpi-serialization/libsqlite3.so, may conflict with libmpi.so.40
Install file: "build/src/release/linux/4.18/64/x86/gcc/8/mpi-serialization/torsional_potential_corrections.mpiserialization.linuxgccrelease" as "bin/torsional_potential_corrections.mpiserialization.linuxgccrelease"
scons: done building targets.

**Comment by roccomoretti** [comment]
You typically only get "scons: done building targets." if the build is successful. So there doesn't look to be any hard errors with the compilation process.

The message you're seeing about conflicts is a warning message. It may or may not be an issue when you actually run things.

If it is an issue, then the most likely cause is the result of partial recompilation in different MPI environments. If you compile part of Rosetta under one MPI system and then compile the rest under a different MPI system, you may see similar such errors. To fix it you'll have to do a clean recompile. Running the (exact same) scons command with `--clean` should remove all the files and allow you to recompile from scratch. Failing that, just deleting the `source/build` directory will also work, though that will delete the files for all compilations.

When you recompile, make sure you're using a consistent MPI environment. For many HPC cluster systems, you may need to do particular `module load` command to establish the proper MPI system. If your compilation gets interrupted and you want to restart it, you may need to re-run the MPI environment setup commands if you're not in the same terminal. Also note that if you submit a job to the HPC queuing system, you'll likely have to include those module loading command within the batch file, as simply activating them prior to submitting the job isn't enough.

**Reply by Mikukub** [reply]
i install openmpi locally and get new error, here after I compile the program and run

this is log file for build
mpiCC -o build/src/release/linux/4.18/64/x86/gcc/8/mpi-serialization/simple_cycpep_predict.mpiserialization.linuxgccrelease -Wl,--disable-new-dtags -Wl,-rpath=/data1/tlarnd2/software/rosetta/source/build/src/release/linux/4.18/64/x86/gcc/8/mpi-serialization -Wl,-rpath=/data1/tlarnd2/software/rosetta/source/build/external/release/linux/4.18/64/x86/gcc/8/mpi-serialization -Wl,-rpath=\$ORIGIN -Wl,-rpath=\$ORIGIN/../lib build/src/release/linux/4.18/64/x86/gcc/8/mpi-serialization/apps/public/cyclic_peptide/simple_cycpep_predict.o -Lexternal/lib -Lbuild/src/release/linux/4.18/64/x86/gcc/8/mpi-serialization -Lsrc -Lbuild/external/release/linux/4.18/64/x86/gcc/8/mpi-serialization -Lexternal -L/data1/tlarnd2/complier/lib -L/usr/local/amber22/lib -L/usr/local/cuda/lib -L/usr/local/cuda/lib64 -L/usr/local/amber22/lib -L/usr/local/cuda/lib -L/usr/local/cuda/lib64 -L/usr/local/amber22/lib -L/usr/local/cuda/lib -L/usr/local/cuda/lib64 -L/usr/local/amber22/lib -L/usr/local/cuda/lib -L/usr/local/cuda/lib64 -L/usr/lib -L/usr/local/lib -ldevel -lprotocols.8 -lprotocols.7 -lprotocols_e.6 -lprotocols_d.6 -lprotocols_c.6 -lprotocols_b.6 -lprotocols_a.6 -lprotocols_h.5 -lprotocols_g.5 -lprotocols_f.5 -lprotocols_e.5 -lprotocols_d.5 -lprotocols_c.5 -lprotocols_b.5 -lprotocols_a.5 -lprotocols.4 -lprotocols.3 -lprotocols_b.2 -lprotocols_a.2 -lprotocols.1 -lcore.6 -lcore.5 -lcore.4 -lcore.3 -lcore.2 -lcore.1 -lbasic -lnumeric -lutility -lObjexxFCL -lz -lzlib -lcppdb -lsqlite3 -lcifparse -lxml2 -lzmq -lrdkit -lbcl -lcmaes
Install file: "build/src/release/linux/4.18/64/x86/gcc/8/mpi-serialization/simple_cycpep_predict.mpiserialization.linuxgccrelease" as "bin/simple_cycpep_predict.mpiserialization.linuxgccrelease"
it seem build success 

but when I  try to run program 

/data1/tlarnd2/software/rosetta/source/bin/simple_cycpep_predict.mpiserialization.linuxgccrelease: symbol lookup error: /data1/tlarnd2/software/rosetta/source/build/src/release/linux/4.18/64/x86/gcc/8/mpi-serialization/libcore.2.so: undefined symbol: _ZN9CifString12UnknownValueB5cxx11E
/data1/tlarnd2/software/rosetta/source/bin/simple_cycpep_predict.mpiserialization.linuxgccrelease: symbol lookup error: /data1/tlarnd2/software/rosetta/source/build/src/release/linux/4.18/64/x86/gcc/8/mpi-serialization/libcore.2.so: undefined symbol: _ZN9CifString12UnknownValueB5cxx11E
/data1/tlarnd2/software/rosetta/source/bin/simple_cycpep_predict.mpiserialization.linuxgccrelease: symbol lookup error: /data1/tlarnd2/software/rosetta/source/build/src/release/linux/4.18/64/x86/gcc/8/mpi-serialization/libcore.2.so: undefined symbol: _ZN9CifString12UnknownValueB5cxx11E
/data1/tlarnd2/software/rosetta/source/bin/simple_cycpep_predict.mpiserialization.linuxgccrelease: symbol lookup error: /data1/tlarnd2/software/rosetta/source/build/src/release/linux/4.18/64/x86/gcc/8/mpi-serialization/libcore.2.so: undefined symbol: _ZN9CifString12UnknownValueB5cxx11E
--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
--------------------------------------------------------------------------
/data1/tlarnd2/software/rosetta/source/bin/simple_cycpep_predict.mpiserialization.linuxgccrelease: symbol lookup error: /data1/tlarnd2/software/rosetta/source/build/src/release/linux/4.18/64/x86/gcc/8/mpi-serialization/libcore.2.so: undefined symbol: _ZN9CifString12UnknownValueB5cxx11E
--------------------------------------------------------------------------
mpirun detected that one or more processes exited with non-zero status, thus causing
the job to be terminated. The first process to do so was:

  Process name: [[54236,1],0]
  Exit code:    127
--------------------------------------------------------------------------

**Comment by Mikukub** [comment, marked as answer]
Problem is fixed by not exporting amber. it seems amber caused a conflict in the machine

**Reply by gongmeiyuan** [reply]
hi, how to fix the bug? remove amber?

**Reply by Mikukub** [reply]
Yes that very stupid, the exxact was build with amber and it cause conflict

**Comment by yrq3027** [comment]
Hi, I met similar issues. I have used Rosetta some time. Today, when I run cartesian_ddg.mpi.linuxgccrelease to calculate ddG, this error thrown suddenly. Can you give some advice to address this issue?
cartesian_ddg.mpi.linuxgccrelease: symbol lookup error: /data/ROSETTA/rosetta/main/source/build/src/release/linux/4.15/64/x86/gcc/7/mpi/libcore.2.so: undefined symbol: _ZN9CifString12UnknownValueB5cxx11E
